standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -6952,8 +6953,8 @@ public OpenTxnsResponse open_txns(OpenTxnRequest rqst) throws TException {
       OpenTxnsResponse response = getTxnHandler().openTxns(rqst);
       List<Long> txnIds = response.getTxn_ids();
       if (txnIds != null && listeners != null && !listeners.isEmpty()) {
-        MetaStoreListenerNotifier.notifyEvent(listeners, EventType.OPEN_TXN,
-            new OpenTxnEvent(txnIds, this));
+        MetaStoreListenerNotifier.notifyTxnEvent(listeners, EventType.OPEN_TXN,
```

Comment:
Shall use original notify() method itself if last 2 parameters are null. Also, add implementation for the same.

Reply:
revert back this call to notifyevent . change notifyTxnEvent to notifyEventDirectSql and change the interface name also